hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java

```diff
@@ -26,26 +31,65 @@
+     * the metadata required to, at minimum, validate the signatures of all required signing keys.
      *
-     * @param txBody the {@link TransactionBody} with the transaction data
+     * @param txn the {@link TransactionBody} with the transaction data
      * @param payer the {@link AccountID} of the payer
+     * @param keyLookup the {@link AccountKeyLookup} to use for key resolution
      * @return the {@link TransactionMetadata} with all information that needs to be passed to
      *     {@link #handle(TransactionMetadata)}
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public TransactionMetadata preHandle(
-            @NonNull final TransactionBody txBody, @NonNull final AccountID payer) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final TransactionBody txn,
+            @NonNull final AccountID payer,
+            @NonNull final AccountKeyLookup keyLookup,
+            @NonNull final ReadableScheduleStore scheduleStore) {
+        final var op = txn.getScheduleDelete();
+        final var id = op.getScheduleID();
+
+        // check for INVALID_SCHEDULE_ID
+        final var scheduleLookupResult = scheduleStore.get(id);
+        if (scheduleLookupResult.isEmpty()) {
+            return new InvalidTransactionMetadata(txn, payer, INVALID_SCHEDULE_ID);
+        }
+
+        // No need to check for SCHEDULE_PENDING_EXPIRATION, SCHEDULE_ALREADY_DELETED,
+        // SCHEDULE_ALREADY_EXECUTED
+        // if any of these are the case then the scheduled tx would not be present in scheduleStore
+
+        // check whether schedule was created with an admin key
+        // if it wasn't, the schedule can't be deleted
+        final var adminKey = scheduleLookupResult.get().adminKey();
+        if (adminKey.isEmpty()) {
+            return new InvalidTransactionMetadata(txn, payer, SCHEDULE_IS_IMMUTABLE);
+        }
+
+        // check whether payer key for this tx matches the admin key for underlying scheduled
+        // transaction
+        final KeyOrLookupFailureReason payerKeyOrLookupFailure = keyLookup.getKey(payer);
+        if (payerKeyOrLookupFailure.failed()) {
+            return new InvalidTransactionMetadata(
+                    txn, payer, payerKeyOrLookupFailure.failureReason());
+        } else {
+            var payerKey = payerKeyOrLookupFailure.key();
+            if (!adminKey.get().equals(payerKey))
+                return new InvalidTransactionMetadata(
+                        txn,
+                        payer,
+                        INVALID_SIGNATURE); // I'm not sure whether this is the correct code
+            // no need to add adminKey to required keys b/c it's also the payer key so already there
+        }
+
+        final var meta =
+                new SigTransactionMetadataBuilder(keyLookup).txnBody(txn).payerKeyFor(payer);
+
```

Comment:
You could add this line at start of method, in parity with other prehandle logics

Reply:
Done.